packages/libs/avector/benches/bench.rs

```diff
@@ -0,0 +1,430 @@
+
+                for _ in 0..iters {
+                    let slab = AVec::<usize>::new();
+                    let bench = MultithreadedBench::<_, 5>::new(Arc::new(slab));
+
+                    let elapsed = bench.run(move |start, slab| {
+                        start.wait();
+
+                        for idx in 0..i {
+                            slab.push(idx);
+                        }
+                    });
+
+                    total += elapsed;
+                }
+
+                total
+            });
+        });
+    }
+}
+
+fn push_single_thread(c: &mut Criterion) {
+    let mut group = c.benchmark_group("push_single_thread");
+
+    for i in N_INSERTIONS {
+        group.bench_with_input(BenchmarkId::new("alloc", i), i, |b, &i| {
+            b.iter_with_setup(Vec::<usize>::new, |mut slab| {
+                for idx in 0..i {
+                    slab.push(idx);
+                }
+            });
+        });
+
+        group.bench_with_input(BenchmarkId::new("intrusive-collections", i), i, |b, &i| {
+            b.iter_with_setup(
+                || intrusive_collections::LinkedList::new(EntryAdapter::new()),
+                |mut slab| {
+                    for idx in 0..i {
+                        let entry = Box::new(Entry {
+                            link: LinkedListLink::new(),
+                            value: Cell::new(idx),
+                        });
+
+                        slab.push_back(entry)
+                    }
+                },
+            );
+        });
+
+        group.bench_with_input(BenchmarkId::new("im", i), i, |b, &i| {
+            b.iter_with_setup(im::Vector::<usize>::default, |mut slab| {
+                for idx in 0..i {
+                    slab.push_back(idx);
+                }
+            });
+        });
+
+        group.bench_with_input(BenchmarkId::new("avector", i), i, |b, &i| {
+            b.iter_with_setup(AVec::<usize>::default, |slab| {
+                for idx in 0..i {
+                    slab.push(idx);
+                }
+            });
+        });
+    }
+}
+
+fn iter_multi_thread(c: &mut Criterion) {
+    let mut group = c.benchmark_group("iter_multi_thread");
+
+    for i in N_INSERTIONS {
+        group.bench_with_input(BenchmarkId::new("alloc", i), i, |b, &i| {
+            b.iter_custom(|iters| {
+                let mut total = Duration::from_secs(0);
+
+                let mut slab = Vec::new();
+                for idx in 0..i {
+                    slab.push(idx);
+                }
+                let slab = RwLock::new(slab);
+                let slab = Arc::new(slab);
+
+                for _ in 0..iters {
+                    let bench = MultithreadedBench::<_, 5>::new(Arc::clone(&slab));
+
+                    let elapsed = bench.run(move |start, slab| {
+                        start.wait();
+
+                        let lock = slab.read().unwrap();
+                        let items: Vec<_> = lock.iter().copied().collect();
```

Comment:
Did you measure the difference by passing the actual data to `black_box` instead of a collected vector? I wonder if `collect()` has an impact on the benchmarks